llvm/docs/DeveloperPolicy.rst

```diff
@@ -336,12 +341,12 @@ you follow these guidelines to help review, search in logs, email formatting
 and so on. These guidelines are very similar to rules used by other open source
 projects.
 
-Most importantly, the contents of the message should be carefully written to
-convey the rationale of the change (without delving too much in detail). It
-also should avoid being vague or overly specific. For example, "bits were not
-set right" will leave the reviewer wondering about which bits, and why they
-weren't right, while "Correctly set overflow bits in TargetInfo" conveys almost
-all there is to the change.
+Commit messages should communicate briefly what the change does, and provide
```

Comment:
IMO, the "why" part is the most important in the commit message. We can understand the "what" from the commit itself, although it is useful to summarize the "what" in the message for the reviewer to check whether the code actually does what the author intended.